### PR TITLE
hotfix/CP-6911-app-banner-copy-to-clipboard-does-not-copying-voucher-code

### PR DESCRIPTION
**Pull request by unnaticleverpush**
App Banner: Copy To Clipboard does not copy voucher code

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/service/NotificationService.java [hunk 1]
```diff
@@ -248,8 +257,17 @@ private NotificationCompat.Builder createBasicNotification(Context context, Stri
 
   private RemoteViews getTextWithImageViews(Context context, Notification notification, boolean hasMedia) {
     RemoteViews expandedView = new RemoteViews(context.getPackageName(), R.layout.notification_text_image_layout);
-    expandedView.setTextViewText(R.id.notification_title, notification.getTitle());
-    expandedView.setTextViewText(R.id.notification_text, notification.getText());
+    String title = notification.getTitle();
+    String text = notification.getText();
+    String voucherCode = notification.getVoucherCode();
+    if (voucherCode != null && title.contains("{voucherCode}")) {
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/NotificationService.java:263]: Move to util function. duplicated code.

**Review by gbhrdt — CHANGES_REQUESTED** [review]

**Review by cpKevinrodriguesnobre — APPROVED** [review]

**Review by iboruch — APPROVED** [review]

File: cleverpush/src/main/java/com/cleverpush/service/NotificationService.java [hunk 2]
```diff
@@ -562,4 +571,16 @@ private static Bitmap loadImageFromDisc(Context context, String fileName) {
 
     return bitmap;
   }
+
+  public String voucherCodeString(String text, String voucherCode) {
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/NotificationService.java:575]: Duplicated code. Maybe put into some Utils class

File: cleverpush/src/main/java/com/cleverpush/service/NotificationService.java [hunk 3]
```diff
@@ -562,4 +571,16 @@ private static Bitmap loadImageFromDisc(Context context, String fileName) {
...
+    try {
+      if (voucherCode != null && !voucherCode.isEmpty() && text.contains("{voucherCode}")) {
+        text = text.replace("{voucherCode}", voucherCode);
+      }
+      return text;
+    } catch (Exception e) {
+      e.printStackTrace();
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/service/NotificationService.java:582]: Never use e.printStackTrace, use Log.e with out Tag

File: cleverpush/src/main/java/com/cleverpush/service/NotificationService.java (hunk 2 above)
Comment (gbhrdt) [reply cleverpush/src/main/java/com/cleverpush/service/NotificationService.java:575]: Also use a more speaking function name. e.g. replaceVoucherCodeString

**Review by gbhrdt — CHANGES_REQUESTED** [review]

**Review by cpKevinrodriguesnobre — APPROVED** [review]

**Pull request merged**